### PR TITLE
14.0 add calendar base booking

### PR DESCRIPTION
**Pull request by sebastienbeau**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Very great addition! Where do you plan to apply it?

cc @alexey-pelykh

**Comment by sebastienbeau** [comment]
I plan to create a module for booking slot for delivery in physical shop.
Then I will have an additional module for using it in shopinvader.

**Comment by pedrobaeza** [comment]
Very great then. I was thinking on task planning our resources allocation, but the same case can be used indeed. Thanks for it.

File: calendar_base_booking/models/bookable_mixin.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+# booked_slot is a slot already booked
+# bookable_slot is a slot (with a size if slot_duration) that fit into
+# an available slot
+
+
+class BookableMixin(models.AbstractModel):
+    _name = "bookable.mixin"
+    _description = "Bookable Mixin"
+
+    slot_duration = fields.Float()
```
Comment (hparfr) [inline calendar_base_booking/models/bookable_mixin.py:25]: uom?

File: calendar_base_booking/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "author": "Akretion, Odoo Community Association (OCA)",
+    "maintainers": ["sebastienbeau"],
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "external_dependencies": {"python": [], "bin": []},
+    "depends": ["calendar"],
+    "data": [],
+    "demo": [],
+    "qweb": [],
```
Comment (hparfr) [inline calendar_base_booking/__manifest__.py:17]: useless keys

File: calendar_base_booking/readme/DESCRIPTION.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Abstract module for support of bookable slot with capacity.
```
Comment (hparfr) [inline calendar_base_booking/readme/DESCRIPTION.rst:1]: with fixed capacity

File: calendar_base_booking/tests/test_bookable.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+from odoo.tests import SavepointCase
+
+CALENDAR = [
+    ("2020-04-06 08:00:00", "2020-04-06 12:00:00"),
+    ("2020-04-06 14:00:00", "2020-04-06 18:00:00"),
+    ("2020-04-07 08:00:00", "2020-04-07 12:00:00"),
+    ("2020-04-07 14:00:00", "2020-04-07 18:00:00"),
+]
+
+# le get_bookable_slot est appeler sur quel object
```
Comment (hparfr) [inline calendar_base_booking/tests/test_bookable.py:18]: translate or rm

**Review by hparfr — CHANGES_REQUESTED** [review]

File: calendar_base_booking/models/bookable_mixin.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+                slots += self._build_bookable_slot(open_slot, slot_start, slot_stop)
+        return slots
+
+    def _get_domain_for_current_object(self):
+        return [
+            ("res_model", "=", self._name),
+            ("res_id", "=", self.id),
+        ]
+
+    def _get_domain(self, start, stop):
```
Comment (kevinkhao) [inline calendar_base_booking/models/bookable_mixin.py:115]: We might easily get conflicts with such a generic function name

**Comment by legalsylvain** [comment]
@sebastienbeau did you planned to finish your PR ?

**Pull request closed without merging**